update project using latest seng-boilerplate

larsvanbraam opened the pull request:
Updated to latest bp version. 
This also closes #9

larsvanbraam committed:
update project using latest seng-boilerplate

larsvanbraam committed:
Fix linting issues

jesse-mm commented on example/index.html:
```diff
@@ -0,0 +1,14 @@
+<html lang="en">
+<head>
+	<meta charset="UTF-8">
+	<title>Seng Boilerplate example</title>
```
SengEvent example

larsvanbraam replied:
I'll update this!

jesse-mm commented on src/lib/AbstractEvent.ts:
```diff
@@ -11,36 +11,36 @@ let _callListenerResult = CallListenerResult.NONE;
 	 *  _true_ if [[cancelable]] is true and [[preventDefault]] has been called on this event.
 	 */
-	private _defaultPrevented:boolean = false;
+	private _defaultPrevented: boolean = false;
```
Does the airbnb standard not force a private variable to be prefixed without a _

Also applicable to all the other defined private variables.

larsvanbraam replied:
@jesse-mm The MediaMonks coding standards suggest private variables to be prefixed with the "_", see the [coding standard document](https://docs.google.com/document/d/1D2EPaZqd6bGetsDjdx4EBSDMkNBuahp86UVG9xbFvAk/edit#heading=h.z337ya). I noticed more differences between the coding standards and our airbnb standards though, did we decide on dropping document entirely?

ThaNarie replied:
That doc is a bit old-ish, for newer projects (react, vue) we switched to the air-bnb coding standard to keep everything consistent between our own (both js and ts) projects, and possibly others that will be using our code.

larsvanbraam replied:
So when did we decide what rules to overwrite in the `tslint.json` file? 

I already had some issues with node modules where the import name rule results in errors, see the seng-disposable example in the readme:
```ts
import Disposable from 'seng-disposable';
...
class AsyncThinger extends Disposable {
...
```
this fails with the current default tslint values. ~~Also when doing for loops for example you cannot use the default `itar` live template anymore since it fails in our standards which is kind of annoying~~
Edit: nvm itar is outdated anyway!.

jesse-mm commented on src/lib/EventDispatcher.ts:
```diff
@@ -152,22 +142,19 @@ export default class EventDispatcher extends Disposable implements IEventDispatc
-		if(isDebugMode && this.hasEventListener(eventType, handler, useCapture))
-		{
+		if (isDebugMode && this.hasEventListener(eventType, handler, useCapture)) {
 			// log.warn(trying to add double listener)
```
This isn't logging anything, even when isDebugMode is set to true xD

jesse-mm commented on src/lib/EventDispatcher.ts:
```diff
@@ -81,43 +79,35 @@ export default class EventDispatcher extends Disposable implements IEventDispatc
-		{
+	public dispatchEvent(event: IEvent): boolean {
+		if (this.isDisposed()) {
 			// todo: _log.error("Can't dispatchEvent on a disposed EventDispatcher");
```
Removing the TODO and throwing an error should be sufficient.

larsvanbraam replied:
Indeed, i'll update this!

jesse-mm commented on src/lib/EventDispatcher.ts:
```diff
@@ -184,23 +171,17 @@ export default class EventDispatcher extends Disposable implements IEventDispatc
-				const listenerData:EventListenerData = this._listeners[eventType][i];
-				if(listenerData.handler === handler && (typeof useCapture === 'undefined' || useCapture === listenerData.useCapture))
-				{
+		else {
```
This branch is not needed as the previous branches do a return.

jesse-mm commented on src/lib/EventPhase.ts:
```diff
@@ -1,8 +1,7 @@
  * An enum for possible event phases that an event can be in
  */
-const enum EventPhase
-{
```
You could leave out the ENUM value assignment.

larsvanbraam replied:
I'll change it to a typescript enum

ThaNarie replied:
Did you guys consider `const` might be added for a specific reason?

https://stackoverflow.com/questions/28818849/how-do-the-different-enum-variants-work-in-typescript

jesse-mm replied:
Didn't know this difference! But what I meant with my previous comment you could use the computed assignment instead of assigning it yourself.
```
const enum Event {
    EVENT_ONE,
    EVENT_TWO,
}
```

larsvanbraam replied:
Ah didn't even know that! Why output larger code if you don't need the lookup object for the event phase, I'll revert this commit!

jesse-mm commented on tslint.json:
```diff
@@ -1,71 +1,20 @@
+    "no-debugger": 2,
+    "no-console": 1,
+    "prefer-array-literal": [ true, { "allow-type-parameters": true } ],
+    "variable-name": [ true, "allow-pascal-case", "allow-leading-underscore" ],
```
allow-leading-underscore should be removed.

larsvanbraam replied:
leading underscores used to be part of the MediaMonks coding standards, did we decide on dropping this?

jesse-mm commented on src/bundle.ts:
```diff
@@ -0,0 +1,20 @@
+export { default as EventListenerData } from './lib/EventListenerData';
+export { default as CallListenerResult } from './lib/CallListenerResult';
+export { default as AbstractEvent } from './lib/AbstractEvent';
+import './lib/event/CommonEvent';
```
Should CommonEvent be part of the lib? as it doesn't even get exported for public usage.

I rather see this back in the readme as an example.

ThaNarie replied:
The import itself doesn't make much sense indeed

larsvanbraam replied:
I'll remove the CommonEvent entirely and update the README.md example to be sufficient!

ThaNarie replied:
Not sure if I agree with removing the file itself. It's added in the package for convenience, because those type of events are used most of the time. You can import them yourself from the `seng-event/lib` folder if you need it, and it will only get included in your project output when you do that.

A lot of packages include a set of related useful files/plugins/adaptors with their main package.

jesse-mm replied:
Then I would like to propose to add DataEvent as well.

jesse-mm commented on src/bundle.ts:
```diff
@@ -0,0 +1,20 @@
+export { default as AbstractEvent } from './lib/AbstractEvent';
+import './lib/event/CommonEvent';
+import './lib/event/BasicEvent';
+import './lib/util/eventTypeUtils';
```
Do we need these imports here?

ThaNarie replied:
Yes, `bundle.js` should include all files, it's for a non-npm (i.e. browser) build.

larsvanbraam committed:
Rename the example title to SengEvent

larsvanbraam committed:
Throw an error when dispatching on a disposed EventDispatcher
Removed commented out todo log

larsvanbraam committed:
Remove the CommonEvent since the README.md already contains an example on how to create an event.
Update the event example in the README.md
Remove useless imports
Update the EventPhase Enum

larsvanbraam committed:
Update the EventPhase Enum

larsvanbraam committed:
Update test to expect an error on failure!

larsvanbraam committed:
Revert "Update the EventPhase Enum"

This reverts commit c9aa4ba

larsvanbraam committed:
Remove leading underscores from private variables

larsvanbraam committed:
Remove self assigned values of the eventPhase enum

larsvanbraam committed:
Remove the reset of the prefixe private variables
Change the import name to work the the default tslint file

larsvanbraam committed:
Resolve the no-increment-decrement rule issues

larsvanbraam committed:
Revert "Remove the CommonEvent since the README.md already contains an example on how to create an event. Update the event example in the README.md Remove useless imports Update the EventPhase Enum"

This reverts commit 85dcfc8

larsvanbraam committed:
Remove the imports in the index.ts
Restore the CommonEvent

larsvanbraam committed:
Update the version number
